client/spec/search_spec.js

```diff
@@ -40,6 +40,33 @@ describe('Search', function() {
         expect(element.all(by.repeater('parameter in tags.selectedKeywords')).count()).toBe(1);
     });
 
+    it('can search by subject codes field', function () {
+        workspace.switchToDesk('SPORTS DESK').then(content.setListView);
+        expect(element.all(by.repeater('items._items')).count()).toBe(2);
+
+        var filterPanelButton = element(by.css('.fitler-trigger'));
+        var subject = element.all(by.css('.dropdown-nested')).first();
+        var subjectToggle = subject.element(by.css('.dropdown-toggle'));
+
+        filterPanelButton.click().then(
```

Comment:
imo it would be better to avoid `then` calls, it should work same like:
```
filterPanelButton.click()
element.all(by.css('[ng-click="toggleModule()"]')).first().click();
subjectToggle.click();
subject.all(by.css('.nested-toggle')).first().click();
subject.all(by.repeater('term in activeTree')).first().click();

expect(element.all(by.repeater('item in items._items')).count()).toBe(0);
expect(element.all(by.repeater('parameter in tags.selectedParameters')).count()).toBe(1);
expect(element.all(by.repeater('t in item[field]')).count()).toBe(1);
```
protractor should handle the promises